### ISSUE TITLE
[Feature]: Support extra args from Network.setUserAgentOverride

### ISSUE THREAD

**Issue by liamdalg** [puppeteer/puppeteer#12688]
### Feature description

`Page.setUserAgent` only supports the `userAgent` and `userAgentMetadata` args from [Network.setUserAgentOverride](https://vanilla.aslushnikov.com/?Network.setUserAgentOverride). Would it be possible to get support for the `platform` and `acceptLanguage` args too? Although it's possible to set these with a raw `CDPSession`, it'd be nice to have it supported by Puppeteer so it  easily propagates to attached CDP sessions.

I'm keen on making a PR for this - curious whether you'd prefer to have the args added to the `Page.setUserAgent` function, or as separate `Page.setPlatform` and `Page.setAcceptLanguage` functions. The former feels preferable, but could be a breaking change.

**Comment by OrKoN** [comment]
I saw that https://developer.mozilla.org/en-US/docs/Web/API/Navigator/platform is deprecated so I am not sure we want to add it: is it reflected in the user agent/request headers in any way? As for acceptLanguage, I think it makes sense but if you are only interested in the accept language headers we have https://pptr.dev/api/puppeteer.page.setextrahttpheaders/

Could you add some details about your use case?

**Comment by liamdalg** [comment]
Fair enough that it's deprecated. The use-case is just that `navigator.platform` doesn't match the user agent and `userAgentData.platform` which some sites can check for. Currently I'm manually setting it and listening for `sessionattached` to propagate it to any new clients, but that feels like it gets in the way of Puppeteer.

As for `acceptLanguage`, I'm interested as I believe it's the only thing that overrides `navigator.language`

```js
import puppeteer from "puppeteer";

const browser = await puppeteer.launch();
const context = await browser.createBrowserContext();
const page = await context.newPage();
const client = await page.createCDPSession();

await client.send("Network.setUserAgentOverride", {
  userAgent: "foo",
  acceptLanguage: "fr-FR",
});

const language = await page.evaluate(() => navigator.language);

# logs fr-FR
console.log(language);
```

**Comment by OrKoN** [comment]
I see, so we currently have an example of doing it using `page.evaluateOnNewDocument` https://pptr.dev/api/puppeteer.page.evaluateonnewdocument/#example The problem with the acceptLanguage in setUserAgentOverride is t it requires you to also override the user agent so you cannot use it independently from the user agent override. Perhaps it would be best to add an implementation for `page.setLanguage(language, languages = [language])` using the evaluateOnNewDocument feature.

**Comment by OrKoN** [comment]
Or perhaps we can add a way to override arbitrary navigator properties? but then evaluateOnNewDocument does not require so much code to do that already.

**Comment by liamdalg** [comment]
I agree that the arbitrary overrides is unnecessary. As for the language/platform issue, `evaluateOnNewDocument` doesn't work for me due to #12706 (as it currently stands)

**Comment by Kle0s** [comment]
According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/platform), it seems like `navigator.platform` is no longer deprecated (?).

If the issue is relevant, I will gladly add this feature.

**Comment by OrKoN** [comment]
Indeed! Let's discuss the proposed design in more details first and I would be happy to review the PR!

**Comment by Kle0s** [comment]
I have two different designs in mind:
1. Simple and straightforward - Enrich `Page.setUserAgent` with a `platform` option: 
`setUserAgent(userAgent: string, userAgentMetadata?: Protocol.Emulation.UserAgentMetadata, platform?: string): Promise<void>;`
In my opinion, it is not really user-friendly, since users will be forced to pass `undefined` for the `userAgentMetadata` argument just to pass anything for the `platform` one.
2. `Page` will now have another method - `setPlatform`. The CDP implementation will be based on keeping state (what we want to have as `userAgent`, `userAgentMetadata` and `platform`) and every call to one of them will actually use `Page.setUserAgent` with all of the relevant information. This requires keeping a state but is user friendly in my opinion.

LMK if you have any notes

**Comment by OrKoN** [comment]
So what would happen if setPlatform is called without any user agent set?

**Comment by Kle0s** [comment]
I thought we would fetch the current user agent (using something like `this.evaluate(() => navigator.userAgent)`) and set it.

**Comment by OrKoN** [comment]
I think we should could use https://chromedevtools.github.io/devtools-protocol/tot/Browser/#method-getVersion to read the current user agent and I think we already call it somewhere. I think thinking if might be better to add an overload to user agent to accept either a string as the first argument or an object with `{userAgent?, userAgentMetadata?, platform?, acceptLanguage?}`?

**Comment by OrKoN** [comment]
eventually we can deprecate the current positional user agent configuration. WDYT?

**Comment by Kle0s** [comment]
Sure, I will take care of that 😄
